src/legacy/ui/public/vislib/visualizations/point_series/column_chart.js

```diff
@@ -167,17 +188,43 @@ export class ColumnChart extends PointSeries {
       if (d.y0 === 0 && yMin > 0) {
         return yScale(yMin) - yScale(d.y);
       }
-
       return Math.abs(yScale(d.y0) - yScale(d.y0 + d.y));
     }
 
+    function formatValue(d) {
+      return chartData.yAxisFormatter(d.y);
+    }
+
     // update
     bars
       .attr('x', isHorizontal ? x : y)
       .attr('width', isHorizontal ? widthFunc : heightFunc)
       .attr('y', isHorizontal ? y : x)
       .attr('height', isHorizontal ? heightFunc : widthFunc);
 
+    const layer = d3.select(bars[0].parentNode);
+    const barLabels = layer.selectAll('text').data(chartData.values.filter(function (d) {
+      return !_.isNull(d.y);
+    }));
+
+    if (isLabels) {
+      const colorFunc = this.handler.data.getColorFunc();
+      const d3Color = d3.rgb(colorFunc(chartData.label));
+      const labelColor = isColorDark(d3Color.r, d3Color.g, d3Color.b) ? 'white' : 'black';
+
+      barLabels
+        .enter()
+        .append('text')
+        .text(formatValue)
+        .attr('x', isHorizontal ? labelX : labelY)
+        .attr('y', isHorizontal ? labelY : labelX)
+        .attr('dominant-baseline', 'central')
+        .attr('text-anchor', 'middle')
+        .attr('font-size', '8pt')
```

Comment:
you can include the font-size and other styles into the label class to reuse the same fontsize as other elements on the chart